rust/worker/src/blockstore/arrow/provider.rs

```diff
@@ -169,61 +171,54 @@ impl BlockManager {
                         let res = bytes.read_to_end(&mut buf).await;
                         match res {
                             Ok(_) => {}
-                            Err(_) => {
-                                // TODO: log error
+                            Err(e) => {
+                                tracing::error!("Error reading block from storage: {}", e);
                                 return None;
                             }
                         }
                         let block = Block::from_bytes(&buf, *id);
                         match block {
                             Ok(block) => {
-                                self.read_cache.write().insert(*id, block.clone());
+                                self.block_cache.insert(*id, block.clone());
                                 Some(block)
                             }
-                            Err(_) => {
-                                // TODO: log error
+                            Err(e) => {
+                                tracing::error!("Error turning bytes into block: {}", e);
                                 None
                             }
                         }
                     }
-                    Err(_) => {
-                        // TODO: log error
+                    Err(e) => {
+                        tracing::error!("Error reading block from storage: {}", e);
                         None
                     }
                 }
             }
         }
     }
 
-    pub(super) async fn flush(&self, id: &Uuid) -> Result<(), Box<dyn ChromaError>> {
-        let block = self.get(id).await;
-
-        match block {
-            Some(block) => {
-                let bytes = match block.to_bytes() {
-                    Ok(bytes) => bytes,
-                    Err(e) => {
-                        return Err(Box::new(e));
-                    }
-                };
-
-                let key = format!("block/{}", id);
-                let res = self.storage.put_bytes(&key, bytes).await;
-                match res {
-                    Ok(_) => {
-                        println!("Block: {} written to storage", id);
-                        Ok(())
-                    }
-                    Err(e) => {
-                        println!("Error writing block to storage {}", e);
-                        Err(Box::new(e))
-                    }
+    pub(super) async fn flush(&self, block: &Vec<Block>) -> Result<(), Box<dyn ChromaError>> {
```

Comment:
I don't understand the utility of this change being &Vec<Block> now? I would prefer if this took a single block if we don't want it to take an id.

**Id -> Block**
The reason to use the id of the block for flushing is to ensure THIS manager is the one tracking the block. A downside now is someone can create blocks outside of the block manager and flush them using this method. It is perfectly reasonable for this method to cache flushed blocks, but since they were created outside of this manager, it could lead to buggy code. 

**Multiple blocks**
I don't think we want this method to take multiple blocks. Firstly, it makes the ergonomics of flushing a single block more cumbersome as we have to wrap it in a vec. Second, we aren't flushing in parallel here. I would rather this flush a single block and then the Blockfile flusher used a futures.all like method to flush all the blocks AND the sparse index in parallel, I think thats cleaner and has the desired performance

Reply:
Any thoughts here?